### PR TITLE
Polymer 3 upgrade

### PR DESCRIPTION
**Pull request by awikkerink**
The main issue I ran into when running the conversion is the need to convert tabs into spaces due to https://github.com/Polymer/polymer-modulizer/issues/441

### CODE REVIEW

File: polymer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "lint": {
-    "rules": ["polymer-2-hybrid"]
+    "rules": ["polymer-2"]
```
Comment (ryantmer) [inline polymer.json]: Is there no `polymer-3` ruleset yet?

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,2 @@
-bower_components*
```
Comment (ryantmer) [inline .gitignore]: 🙌

File: package.json [hunk 3]
```diff
@@ -1,32 +1,46 @@
...
   "license": "Apache-2.0",
   "devDependencies": {
+    "@webcomponents/webcomponentsjs": "^2.0.2",
+    "babel-eslint": "^8.2.3",
     "eslint": "^4.15.0",
     "eslint-config-brightspace": "^0.4.0",
     "eslint-plugin-html": "^4.0.1",
-    "polymer-cli": "^1.5.7"
+    "polymer-cli": "^1.7.3",
+    "wct-browser-legacy": "^1.0.1"
```
Comment (dlockhart) [inline package.json]: Why are they calling it "legacy"? Is there a new WCT they're working on that isn't ready yet?

Comment (ryantmer) [reply package.json]: I was also confused by this - couldn't figure that out with 60 or so seconds of googling 😐

**Comment by dlockhart** [comment]
Did you use Yarn? Reading through their [hands-on guide](https://www.polymer-project.org/blog/2017-08-23-hands-on-30-preview), that's what they recommend. And you end up with a `flat: true` in `package.json` to ensure that all the NPM dependencies are at the root.

File: polymer.json (hunk 1 above)
Comment (awikkerink) [reply polymer.json]: There is. I didn't know when I did the d2l-image conversion

File: package.json (hunk 3 above)
Comment (awikkerink) [reply package.json]: It's part of the modulizer conversion, and if this package doesn't exist, `polymer test` fails with a mysterious error.

**Comment by awikkerink** [comment]
@dlockhart no I didn't use yarn. The dependency graph is flat for `@polymer` dependencies.

It's probably more important to use yarn at the application level.

**Pull request merged**